plugins/state_history_plugin/state_history_plugin.cpp

```diff
@@ -65,12 +68,16 @@ struct state_history_plugin_impl : std::enable_shared_from_this<state_history_pl
    block_id_type      lib_id;
    time_point         head_timestamp;
 
-   constexpr static uint64_t default_frame_size = 1024 * 1024;
-
    named_thread_pool<struct ship> thread_pool;
 
    bool  plugin_started = false;
 
+public:
+   void plugin_initialize(const variables_map& options);
+   void plugin_startup();
+   void plugin_shutdown();
+   session_manager& get_session_manager() { return session_mgr; }
```

Comment:
I find it so very annoying that gcc does not like `session_manager`. I have ran into that so many times. I like `session_manager()` over `get_`. Oh well :(

Reply:
haha yes! I was surprised as well.